doc/architecture/decisions/0009-primary-person.md

```diff
@@ -0,0 +1,58 @@
+# 9. Primary Person
+
+## Context
+
+We need to be able to identify/uniquely handle responses and the routing path of the primary person within a household. Currently people within a household would be represented as a list, which generates a number of pages distinct from the main routing path.
+
+## Decision
+
+### Add context to lists
+
+Context allows storage of additional information related to lists. Each list collection can store with it a map of context for the list. For example, the primary householder would be represented as follows:
+
+```json
+{
+    "answers": [
+        {
+            "answer_id": "date-of-birth",
+            "value": "1982-08-01",
+            "list_item_id": "a9hd8j"
+        },
+        {
+            "answer_id": "occupation",
+            "value": "Teacher",
+            "list_item_id": "a9hd8j"
+        }
+    ],
+    "lists": [
+        {
+            "name": "people",
+            "context": {
+                "primary_person": "a9hd8j"
+            },
+            "items": [
+                "a9hd8j"
+            ]
+        }
+    ]
+}
+```
+
+### Create a PrimaryPersonCollector block type
+
+A new block type "PrimaryPersonCollector" will be defined to support the necessary behaviour. It will extend the existing ListCollector and support adding a primary person through adding context to the list store. The following diagram identifies the flow to be implemented in runner.  
+
+![Primary Flow](0009-primary-flow.png)
+
+Both the PrimaryPersonListCollector and ListCollector operate upon the same list in the store. The PrimaryPersonCollector would create an item in the list store, before defining any answers it refers to. This allows the identification of a primary person in one block and deferring their name entry to another.
+
+On routing backwards from the ListCollector, a custom route handler would be implemented to prevent the primary person reaching add behaviour. This is represented in the reverse flow below.
```

Comment:
This doesn't make sense. The routing path wouldn't include the add screen as it would be defined as part of the `PrimaryPersonListCollector`, so I assume this is referring to the back button behaviour? If this is the case, if the url for the primary person add/edit screen has the list item id in it, we should be safe to display it?

Reply:
I thought the point needed to be made here that navigating using previous/back button to the add screen should result in the driving question being shown?

Reply:
I don't think so, we shouldn't break the default back button behaviour.